graphql/resolvers.go

```diff
@@ -3266,10 +3252,23 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 		}
 	}
 
-	for _, v := range versions {
-		if activatedVersionCount == limit {
+	index := 0
+	versionsCheckedCount := 0
+	for matchingVersionCount < limit {
+		// If we no longer have any more versions to check break out and return what we have
+		if index >= len(versions) {
 			break
 		}
+		// If we have checked more versions then the MaxMainlineCommitVersionLimit then break out and return what we have
+		if versionsCheckedCount >= model.MaxMainlineCommitVersionLimit {
+			// Return an error if we did not find any versions that match
+			if matchingVersionCount == 0 {
+				return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting activated versions: exceeded max limit of %d versions", model.MaxMainlineCommitVersionLimit))
```

Comment:
This error won't really make sense to users. Maybe "matching version not found in %d most recent versions" ? But also I think this case would get caught by 3259, right? If versions is empty even the first loop we'd break at 3259.